modular/blocksyncer/blocksyncer_indexer.go

```diff
@@ -1,551 +0,0 @@
-			err := eventModule.HandleEvent(ctx, block, txHash, event)
-			if err != nil {
-				log.Errorw("failed to handle event", "module", module.Name(), "event", event, "error", err)
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-func (i *Impl) ExtractEvent(ctx context.Context, block *coretypes.ResultBlock, txHash common.Hash, event sdk.Event) (interface{}, error) {
-	for _, module := range i.Modules {
-		if eventModule, ok := module.(modules.EventModule); ok {
-			data, err := eventModule.ExtractEvent(ctx, block, txHash, event)
-			if err != nil {
-				return nil, err
-			}
-			if data != nil {
-				return data, nil
-			}
-		}
-	}
-	return nil, nil
-}
-
-// Process fetches a block for a given height and associated metadata and export it to a database.
-// It returns an error if any export process fails.
-func (i *Impl) Process(height uint64) error {
-	// log.Debugw("processing block", "height", height)
-	var block *coretypes.ResultBlock
-	var events *coretypes.ResultBlockResults
-	var txs []*types.Tx
-	var err error
-	heightKey := fmt.Sprintf("%s-%d", i.GetServiceName(), height)
-	blockAny, okb := blockMap.Load(heightKey)
-	eventsAny, oke := eventMap.Load(heightKey)
-	txsAny, okt := txMap.Load(heightKey)
-	block, _ = blockAny.(*coretypes.ResultBlock)
-	events, _ = eventsAny.(*coretypes.ResultBlockResults)
-	txs, _ = txsAny.([]*types.Tx)
-	if !okb || !oke || !okt {
-		log.Warnf("failed to get map data height: %d", height)
-		return ErrBlockNotFound
-	}
-
-	startTime := time.Now().UnixMilli()
-
-	beginBlockEvents := events.BeginBlockEvents
-	endBlockEvents := events.EndBlockEvents
-	txCount := len(txs)
-	eventCount := 0
-
-	// 1. handle events in startBlock
-
-	if len(beginBlockEvents) > 0 {
-		eventCount += len(beginBlockEvents)
-		err = i.ExportEventsWithoutTx(context.Background(), block, beginBlockEvents)
-		if err != nil {
-			log.Errorf("failed to export events without tx: %s", err)
-			return err
-		}
-	}
-
-	// 2. handle events in txs
-	err = i.ExportEventsInTxs(context.Background(), block, txs)
-	if err != nil {
-		log.Errorf("failed to export events in txs: %s", err)
-		return err
-	}
-
-	// 3. handle events in endBlock
-	if len(endBlockEvents) > 0 {
-		eventCount += len(endBlockEvents)
-		err = i.ExportEventsWithoutTx(context.Background(), block, endBlockEvents)
-		if err != nil {
-			log.Errorf("failed to export events without tx: %s", err)
-			return err
-		}
-	}
-
-	err = i.ExportEpoch(block)
-	if err != nil {
-		log.Errorf("failed to export epoch: %s", err)
-		return err
-	}
-
-	log.Infof("handle&write data cost: %d", time.Now().UnixMilli()-startTime)
-	log.Infof("height :%d tx count:%d event count:%d", height, txCount, eventCount)
-
-	blockMap.Delete(heightKey)
-	eventMap.Delete(heightKey)
-	txMap.Delete(heightKey)
-	i.ProcessedHeight = height
-
-	cost := time.Now().UnixMilli() - startTime
-	log.Infof("total cost: %d", cost)
-	metrics.BlocksyncerCatchTime.WithLabelValues(fmt.Sprintf("%d", height)).Set(float64(cost))
```

Comment:
I think this will cause the Prometheus metrics to become extreme large with thus dynamic labels.

Reply:
This has been fixed